pkg/dyninst/decode/types.go

```diff
@@ -778,9 +877,18 @@ func (p *pointerType) encodeValueFields(
 	//
 	// For things like map buckets or channel internals, which we encode as pointers, we won't
 	// find a go kind.
-	goKind, ok := p.Pointee.GetGoKind()
+	goKind, ok := (*ir.PointerType)(p).Pointee.GetGoKind()
 	writeAddress := ok && goKind != reflect.Pointer
-	return encodePointer(data, writeAddress, p.Pointee.GetID(), enc, d)
+	return encodePointer(data, writeAddress, (*ir.PointerType)(p).Pointee.GetID(), enc, d)
+}
+
+func (p *pointerType) encodeLiteralValue(
+	d *Decoder,
+	enc *jsontext.Encoder,
+	data []byte,
+) error {
+	// For literal value, we just want the pointed-to value without address
```

Comment:
Is this definitely true? I feel like more design is needed here. Don't we want to eliminate the envelopes all the way down?